File: php/phpsearch/searchresult.php
<?php

/**
 * Class SearchResult
 *
 * @property string pattern
 * @property object file
 * @property int linenum
 * @property int match_start_index
 * @property int match_end_index
 * @property string line
 * @property array lines_before
 * @property array lines_after
 * @property int maxlinelength
 * @property bool colorize
 */
class SearchResult
{
    public function __construct(
        string $pattern,
        $file,
        int $linenum,
        int $match_start_index,
        int $match_end_index,
        string $line,
        array $lines_before,
        array $lines_after
    ) {
        $this->pattern = $pattern;
        $this->file = $file;
        $this->linenum = $linenum;
        $this->match_start_index = $match_start_index;
        $this->match_end_index = $match_end_index;
        $this->line = $line;
        $this->lines_before = $lines_before;
        $this->lines_after = $lines_after;
    }
}

File: java/javasearch/src/main/java/javasearch/SearchResult.java
/*******************************************************************************
SearchResult

Class to encapsulate a command line search option

@author Cary Clark &lt;clarkcb@gmail.com&gt;
@version $Rev$
@copyright Cary Clark 2012
*******************************************************************************/

package javasearch;

import javafind.FileResult;

import java.util.ArrayList;
import java.util.List;
import java.util.regex.Pattern;

public class SearchResult {

    final private Pattern searchPattern;
    private FileResult fileResult;
    final private int lineNum;
    final private int matchStartIndex;
    final private int matchEndIndex;
    final private String line;
    final private List<String> linesBefore;
    final private List<String> linesAfter;

    public SearchResult(final Pattern searchPattern,
                        final FileResult file,
                        final int lineNum,
                        final int matchStartIndex,
                        final int matchEndIndex,
                        final String line) {
        this(searchPattern, file, lineNum, matchStartIndex, matchEndIndex, line,
             new ArrayList<>(), new ArrayList<>());
    }

    public SearchResult(final Pattern searchPattern,
                        final FileResult file,
                        final int lineNum,
                        final int matchStartIndex,
                        final int matchEndIndex,
                        final String line,
                        final List<String> linesBefore,
                        final List<String> linesAfter) {
        this.searchPattern = searchPattern;
        this.fileResult = file;
        this.lineNum = lineNum;
        this.matchStartIndex = matchStartIndex;
        this.matchEndIndex = matchEndIndex;
        this.line = line;
        this.linesBefore = linesBefore;
        this.linesAfter = linesAfter;
    }

    public final Pattern getSearchPattern() {
        return this.searchPattern;
    }

    public final FileResult getFileResult() {
        return this.fileResult;
    }

    protected final void setFileResult(final FileResult sf) {
        this.fileResult = sf;
    }

    public final int getLineNum() {
        return this.lineNum;
    }

    public final int getMatchStartIndex() {
        return this.matchStartIndex;
    }

    public final int getMatchEndIndex() {
        return this.matchEndIndex;
    }

    public final String getLine() {
        return this.line;
    }

    public final List<String> getLinesBefore() {
        return this.linesBefore;
    }

    public final List<String> getLinesAfter() {
        return this.linesAfter;
    }
}
